src/initial_conditions/update_initial_conditions.jl

```diff
@@ -1,3 +1,143 @@
+function _update_initial_conditions!(
+    ic_vector::Vector{T},
+    state::StateInfo,
+    ::Dates.Period,
+) where {
+    T <:
+    InitialCondition{InitialTimeDurationOn, S},
+} where {S <: Union{Float64, PJ.ParameterRef}}
+    for ic in ic_vector
+        sdata = get_state_data(state, TimeDurationOn(), get_component_type(ic))
+        var_val = get_state_values(sdata)
+        set_ic_quantity!(ic, var_val[1, get_component_name(ic)])
+    end
+    return
+end
+
+function _update_initial_conditions!(
+    ic_vector::Vector{T},
+    state::StateInfo,
+    ::Dates.Period,
+) where {
+    T <: InitialCondition{
+        InitialTimeDurationOff,
+        S,
+    },
+} where {S <: Union{Float64, PJ.ParameterRef}}
+    for ic in ic_vector
+        sdata = get_state_data(state, TimeDurationOff(), get_component_type(ic))
+        var_val = get_state_values(sdata)
+        set_ic_quantity!(ic, var_val[1, get_component_name(ic)])
+    end
+    return
+end
+
+function _update_initial_conditions!(
+    ic_vector::Vector{T},
+    state::StateInfo,
+    ::Dates.Period,
+) where {T <: InitialCondition{DevicePower, S}} where {S <: Union{Float64, PJ.ParameterRef}}
+    for ic in ic_vector
+        status_data = get_state_data(state, OnVariable(), get_component_type(ic))
+        status_val = get_state_values(status_data)[1, get_component_name(ic)]
```

Comment:
I think this syntax could be confusing because it will return the value of the state and not the current value of the ic. I made several changes to the syntax that hopefully help clean it up.